tests/optim/test_optim.py

```diff
@@ -75,6 +75,47 @@ def optim_params(module_name, param_name):
         assert fixed_param_unchanged and not free_param_unchanged
 
 
+@pytest.mark.parametrize('scheduler', [optim.LambdaLR,
+                                       optim.StepLR,
+                                       optim.ExponentialLR,
+                                       ])
+def test_dynamic_lr(scheduler):
+    pyro.clear_param_store()
+    gamma = 2
+    step_size = 1
+    optimizer = torch.optim.SGD
+    if scheduler is optim.LambdaLR:
+        pyro_scheduler = scheduler({'optimizer': optimizer, 'optim_args': {'lr': 0.01},
```

Comment:
nit: why not just pass into the parameters themselves:

```
@pytest.mark.parametrize('scheduler', [optim.LambdaLR(..scheduler args..),)
```

or separately as `scheduler_args` if you prefer. That makes the test free of any parametrization logic.

Reply:
i had this originally but it looks really gross to make it pep8 compatible (a lot of line breaks to make it < 120 char). i can do it if you think its cleaner?

Reply:
You could make more space by breaking after the `[`:

```
@pytest.mark.parametrize('scheduler', [
    (param1),
    (param2),
    ..]
```

and each param could take multiple lines, if needed. That'll probably be cleaner, but I'll leave it to your judgement.